XWIKI-22930: Scroll bars appear in the right side of each icon

Sereza7 opened the pull request:
# Jira URL

<!-- Add the link to the corresponding JIRA issue referenced in a commit message. Unless this is a [Misc] commit,
see https://dev.xwiki.org/xwiki/bin/view/Community/DevelopmentPractices#HRule:Don27tcreateunnecessaryissues
-->
https://jira.xwiki.org/browse/XWIKI-22930
For reference, caused by the commit from: https://github.com/xwiki/xwiki-platform/pull/3800


# Changes

## Description

* Updated the overflow-y value so that the scroll control only appears for elements whose name actually overflows.

## Clarifications

<!-- Provide extra hints to make it easier to understand the PR. Those could be:
* Explanation of choices made in this PR
* Anchor towards extra resources needed to understand the context of this PR (e.g., a forum proposal).
* Links to other issues this issue depends on
-->

* On my firefox installation, this style does not make much difference. But on Chrome and other firefox installs, this small change revert the regression on style.

# Screenshots & Video

<!-- If this PR introduces any UI change, it's recommended to highlight it with before/after screenshots 
or even a screen recording for complex interactions. 
-->
Before the changes, on Chrome:
![image](https://github.com/user-attachments/assets/42399f9d-745a-46cd-bb50-4d18e9d6f399)


After the changes, on Chrome:
![image](https://github.com/user-attachments/assets/bc9eae35-0acf-4007-b819-5514439bfc60)
We can see that the scroll indicators are no longer displayed everywhere but only for the names that can be scrolled.
This is the expected behaviour.
# Executed Tests

<!-- Especially important for regression fixes. 
Indicate how changes were tested (e.g., what maven commands were run to validate them).
-->
Manual tests as seen above, tested only on Firefox and Chrome on Debian.
No automatic tests. This regression wasn't caught by any automatic tests and the solution we picked is very specific, it won't impact other UIs.
# Expected merging strategy

* Prefers squash: Yes <!-- No — Explain why. -->
* Backport on branches:
  * 16.10.X (same as https://jira.xwiki.org/browse/XWIKI-22339 )

Sereza7 commented:
Set @surli as the assignee of this blocker because he knows the changes related to this regression:
-> https://github.com/xwiki/xwiki-platform/pull/3800

michitux commented:
> only for the names that can be scrolled.
> This is the expected behaviour.

Is this really expected? Shouldn't the names just be fully displayed? I mean as these are only names in the icon theme definition, they aren't arbitrarily long.

Sereza7 commented:
> Is this really expected? Shouldn't the names just be fully displayed? I mean as these are only names in the icon theme definition, they aren't arbitrarily long.

Users can create their own icon themes and extend the existing ones, so they can become arbitrarily long. Some are really long and the initial UI was meant to fit only the 'simple' names. 
I tried to keep the UI as close to possible as what used to be.
If we make the boxes larger, it would also mean that we lose space efficiency, expecially for 'simple' names where the boxes would be mostly empty.
I think we shouldn't decrease the font-size here.

At least this is the behaviour I expected. 

Note that without those changes brought in XWIKI-22339, the names would just overflow.

___
PS Gabi just told me that she thinks that this does not look good either. I'll open a brainstorming topic on the forum to gather ideas.

michitux commented:
Some ideas:

* make the height of the icon rows dynamic, so only rows with long names need more space.
* abbreviate the icon names using `text-overflow: ellipsis` and display the full name in a tooltip (and make sure it's still fully read in screen readers)

Sereza7 commented:
Thanks for the ideas :)
I think it'd be great to get @tkrieck 's help on this one. All solutions need a compromise and I'm not sure which one would be the best.

tkrieck commented:
> Thanks for the ideas :) I think it'd be great to get @tkrieck 's help on this one. All solutions need a compromise and I'm not sure which one would be the best.

I've achieved a good result by changing the flex layout for a grid one. It seems appropriate, since the icon matrix is in fact a "grid" layout. Also with grid, if one of the icon names breaks into 3 or 4 lines, only that row will be taller than the others

Another layout change is making the button itself a flex layout.

If you want to test the rules here's the updated CSS (the commented properties are what I had to remove to make it work)
```

.xwikiIconPickerList {
    max-width: 100%;
    overflow-y: scroll;
    max-height: 240px;
    gap: .3em;
    display: grid;
    grid-template-columns: repeat(4, 1fr);  /*4 columns, each a fraction of the whole width */ 
}

.xwikiIconPickerIcon {
    /* height:80px; */
    /* width:100px; */
    text-align: center;
    border-radius: 7px;
    vertical-align: middle;
    /* overflow:hidden; */
    /* padding:10px; */
    cursor: pointer;
    background-color: transparent;
    margin: 0;
    padding: 0;
    display: flex;
    flex-flow: column;
}

.xwikiIconPickerIcon .xwikiIconPickerIconName {
    /* max-height:2.9em; */
    /* overflow-y:scroll; */
    /* overflow-x:hidden; */
    white-space:pre-wrap;
    word-break:break-word;
}
```

There's more to change in the icon picker, the "preview with" section in particular is bothering me (bad layout, too much focus there) but that's a subject for another proposal.

Sereza7 commented:
I implemented the style changes proposed by Thiago in https://github.com/xwiki/xwiki-platform/pull/3933/commits/0bd72b30f75465105b4eac93c2a288edb3d0b7c2 with a couple small changes:
* Removed the default margin under the iconName that left some unused space at the bottom of the buttons.
* Added some padding on the side of the grid so that it'd look a bit more consistent (same gap between sides and first item as between the first two items on a line)


Here are the results after those last changes on Chrome and Firefox:
![Screenshot from 2025-03-20 14-26-55](https://github.com/user-attachments/assets/6c2ce7a7-376d-44cf-af7a-18b7875e4887)


![Screenshot from 2025-03-20 14-28-55](https://github.com/user-attachments/assets/8437c709-e569-4782-9e67-c7c8485d7589)

It looks better to me now, and somehow every title fits into its box :)

PR is ready for further review or a merge.

mflorea commented on xwiki-platform-core/xwiki-platform-icon/xwiki-platform-icon-ui/src/main/resources/IconThemesCode/IconPicker.xml:
```diff
@@ -625,11 +625,13 @@ require(['jquery', 'xwiki-icon-picker'], function($) {
+  overflow-y: scroll;
   gap: .3em;
+  padding: .3em;
+  grid-template-columns: repeat(4, 1fr);  /* 4 columns, each a fraction of the whole width */
```
```suggestion
  /* 4 columns, each a fraction of the whole width */
  grid-template-columns: repeat(4, 1fr);
```

mflorea reviewed (approved):
Looks good.
